### PR TITLE
crush: remove last traces of ruleset; remove min/max_size

### PR DESCRIPTION
**Pull request by liewegas**
We made ruleid == ruleset several releases back.  Remove all remaining traces.

Drop teh min/max_size fields that go along with that.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by avanthakkar — CHANGES_REQUESTED** [review]

**Comment by avanthakkar** [comment]
@liewegas  Why the min/max_size fields are dropped? The only concern for me is that the dashboard pool-form component use these fields to display the invalid input message which looks something like this:
Screenshot from 2021-06-30 15-42-25
So are these fields not relevant to show anymore?

**Review by avanthakkar — CHANGES_REQUESTED** [review]
`ruleset` should be removed from here https://github.com/ceph/ceph/blob/c42712dd18032d9651e2ca5f6fb9ae5a078378df/src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.html#L372

**Comment by liewegas** [comment]
> @liewegas Why the min/max_size fields are dropped? The only concern for me is that the dashboard pool-form component use these fields to display the invalid input message which looks something like this:
> Screenshot from 2021-06-30 15-42-25
> So are these fields not relevant to show anymore?

Yeah, no longer relevant.  For replication, I would enforce a range of 1..10.  For EC pools, there is no hard-coded limit, but probably 24 is a reasonable ceiling.

**Comment by avanthakkar** [comment]
> > @liewegas Why the min/max_size fields are dropped? The only concern for me is that the dashboard pool-form component use these fields to display the invalid input message which looks something like this:
> > Screenshot from 2021-06-30 15-42-25
> > So are these fields not relevant to show anymore?
> 
> Yeah, no longer relevant. For replication, I would enforce a range of 1..10. For EC pools, there is no hard-coded limit, but probably 24 is a reasonable ceiling.

Yes, `Replica Size` only exist in case of replicated pool type. So i'd suggest to remove `getMaxSize()` & `getMaxSize()` from ts and html files of pool-form component and instead just hard-code the values for `[max]` & `[min]` here:
```
<!-- Replica Size -->
          <div class="form-group row"
               *ngIf="isReplicated">
            <label class="cd-col-form-label required"
                   for="size"
                   i18n>Replicated size</label>
            <div class="cd-col-form-input">
              <input class="form-control"
                     id="size"
                     [max]="getMaxSize()"
                     [min]="getMinSize()"
```

Also you need to remove all occurrences of rule.max/min_size from pool-form.component.ts and if their values are used maybe replace it with the hard-coded one as you said.

**Review by avanthakkar — APPROVED** [review]

**Comment by liewegas** [comment]
Thanks @avanthakkar !

**Comment by liewegas** [comment]
@avanthakkar one of the dashboard tests failed this time around...

**Comment by avanthakkar** [comment]
> @avanthakkar one of the dashboard tests failed this time around...

The failure doesn't seem to be related to the changes..will try re-triggering the test.

**Comment by avanthakkar** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/controllers/crush_rule.py [hunk 1]
```diff
@@ -11,10 +11,7 @@
 LIST_SCHEMA = {
     "rule_id": (int, 'Rule ID'),
     "rule_name": (str, 'Rule Name'),
-    "ruleset": (int, 'RuleSet related to the rule'),
     "type": (int, 'Type of Rule'),
-    "min_size": (int, 'Minimum size of Rule'),
-    "max_size": (int, 'Maximum size of Rule'),
     'steps': ([{str}], 'Steps included in the rule')
 }
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/controllers/crush_rule.py:17]: @avanthakkar would plz you mind adding the v2.0 to these endpoints? We don't have a decorator to set the version for all the endpoints of a resource, right?

Comment (avanthakkar) [reply src/pybind/mgr/dashboard/controllers/crush_rule.py:17]: Ah yes, will do that ASAP!

Comment (epuertat) [reply src/pybind/mgr/dashboard/controllers/crush_rule.py:17]: thank you!

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Comment by avanthakkar** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/controllers/crush_rule.py [hunk 2]
```diff
@@ -24,9 +21,11 @@
 class CrushRule(RESTController):
     @EndpointDoc("List Crush Rule Configuration",
                  responses={200: LIST_SCHEMA})
+    @RESTController.MethodMap(version='2.0')
     def list(self):
         return mgr.get('osd_map_crush')['rules']
 
+    @RESTController.MethodMap(version='2.0')
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/controllers/crush_rule.py:28]: Thanks a lot, @avanthakkar!

**Review by epuertat — APPROVED** [review]
LGTM! Thanks!

**Comment by liewegas** [comment]
https://pulpito.ceph.com/sage-2021-07-14_00:20:53-rados-wip-sage-testing-2021-07-13-1820-distro-basic-smithi/
https://pulpito.ceph.com/sage-2021-07-14_15:33:57-rados-wip-sage-testing-2021-07-13-1820-distro-basic-smithi/

**Pull request merged**